armi/nuclearDataIO/xsLibraries.py

```diff
@@ -200,8 +200,22 @@ def mergeXSLibrariesInWorkingDirectory(lib, xsLibrarySuffix="", mergeGammaLibs=F
                 for nuc in neutronLibrary.nuclides
                 if isinstance(nuc._base, nuclideBases.DummyNuclideBase)
             ]
+
+            gamisoLibraryPath = nuclearDataIO.getExpectedGAMISOFileName(suffix=xsLibrarySuffix, xsID=xsID)
+            pmatrxLibraryPath = nuclearDataIO.getExpectedPMATRXFileName(suffix=xsLibrarySuffix, xsID=xsID)
+
+            # Check if the gamiso and pmatrx data paths exist with the xs library suffix so that
+            # these are merged in. If they don't both exist then that is OK and we can just
+            # revert back to expecting the files just based on the XS ID.
+            if not (os.path.exists(gamisoLibraryPath) and os.path.exists(pmatrxLibraryPath)):
+                runLog.warning(f"Neither the GAMISO or PMATRX data exist for "
+                               f"XS ID {xsID} with suffix {xsLibrarySuffix}. "
+                               f"Attempting to find GAMISO/PMATRX data with "
+                               f"only XS ID {xsID} instead.")
+                gamisoLibraryPath = nuclearDataIO.getExpectedGAMISOFileName(xsID=xsID)
+                pmatrxLibraryPath = nuclearDataIO.getExpectedPMATRXFileName(xsID=xsID)
```

Comment:
Do we not want to check if these files exist, too?

Reply:
We could, but currently the ISOTXS files try to get merged and then the global flux solver just fails later on if something wasn't found. I kind of like that approach, but there is there weird "do ISOTXS and then do GAMISO/PMATRX similarly, but the code is a bit different to treat the file names and such."